### PR TITLE
ci: add linting workflow

### PR DESCRIPTION
**Pull request by yqlbu**
### Background

Make use of `golang-ci-lint` to harden code quality. Introduce linting workflow and fix some existing warns/errors in the code base.

> **Note**
> Adding `ci-lint` is a good way to enforce quality check to contributors that are outside of @juicity.

### Checklist

- [x] The Pull Request has been fully tested
- [ ] There's an entry in the CHANGELOGS
- [ ] There is a user-facing docs PR against https://github.com/juicity/juicity

### Full changelogs

- ci: add linting workflow
- fix: fix linting warns/errors

### Issue reference

NA

### Test Result

https://github.com/juicity/juicity/actions/runs/5727695886

image

### CODE REVIEW

**Review by mzz2017 — DISMISSED** [review]
LGTM

File: cmd/client/run.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ var (
 				}
 			}()
 			sigs := make(chan os.Signal, 1)
-			signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP, syscall.SIGQUIT, syscall.SIGKILL, syscall.SIGILL)
+			signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP, syscall.SIGQUIT, syscall.SIGTERM, syscall.SIGILL)
```
Comment (mzz2017) [inline cmd/client/run.go:87]: duplicated sigterm

File: cmd/server/run.go [hunk 2]
```diff
@@ -81,7 +81,7 @@ var (
 				}
 			}()
 			sigs := make(chan os.Signal, 1)
-			signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP, syscall.SIGQUIT, syscall.SIGKILL, syscall.SIGILL)
+			signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP, syscall.SIGQUIT, syscall.SIGTERM, syscall.SIGILL)
```
Comment (mzz2017) [inline cmd/server/run.go:84]: duplicated sigterm

Comment (yqlbu) [reply cmd/server/run.go]: Resolved in https://github.com/juicity/juicity/pull/52/commits/a413aac3009144876527d572308549262d6de4a8

File: cmd/client/run.go (hunk 1 above)
Comment (yqlbu) [reply cmd/client/run.go]: Same as below

**Pull request merged**